src/runtime_src/hip/api/hip_event.cpp

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023-2024 Advanced Micro Device, Inc. All rights reserved.
+
+#include "core/common/error.h"
+
+#include "hip/config.h"
+#include "hip/hip_runtime_api.h"
+
+#include "hip/core/event.h"
+
+namespace xrt::core::hip {
+
+void throw_if_null(hipEvent_t event)
```

Comment:
Please write a generic function and keep it in common area sothat all other files can use it.
Not a blocker for merging this change